packages/mgt-components/src/components/mgt-search-results/mgt-search-results.ts

```diff
@@ -0,0 +1,1268 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import { html, HTMLTemplateResult, nothing, TemplateResult } from 'lit';
+import { property, state } from 'lit/decorators.js';
+import {
+  CacheService,
+  CacheStore,
+  equals,
+  MgtTemplatedComponent,
+  prepScopes,
+  Providers,
+  ProviderState,
+  customElement,
+  mgtHtml,
+  BetaGraph,
+  BatchResponse
+} from '@microsoft/mgt-element';
+
+import { schemas } from '../../graph/cacheStores';
+import { strings } from './strings';
+import { styles } from './mgt-search-results-css';
+import {
+  DirectoryObject,
+  Drive,
+  DriveItem,
+  EntityType,
+  List,
+  ListItem,
+  Message,
+  SearchHit,
+  SearchHitsContainer,
+  SearchRequest,
+  SearchResponse,
+  Site
+} from '@microsoft/microsoft-graph-types';
+import { SearchRequest as BetaSearchRequest } from '@microsoft/microsoft-graph-types-beta';
+import {
+  getIsResponseCacheEnabled,
+  getNameFromUrl,
+  getRelativeDisplayDate,
+  getResponseInvalidationTime,
+  sanitizeSummary,
+  trimFileExtension
+} from '../../utils/Utils';
+import { getSvg, SvgIcon } from '../../utils/SvgHelper';
+import { fluentSkeleton, fluentButton, fluentTooltip, fluentDivider } from '@fluentui/web-components';
+import { registerFluentComponents } from '../../utils/FluentComponents';
+import { CacheResponse } from '../CacheResponse';
+
+registerFluentComponents(fluentSkeleton, fluentButton, fluentTooltip, fluentDivider);
+
+/**
+ * Object representing a thumbnail
+ */
+interface Thumbnail {
+  /**
+   * The url of the Thumbnail
+   */
+  url?: string;
+}
+
+/**
+ * Object representing a Binary Thumbnail
+ */
+interface BinaryThumbnail {
+  /**
+   * The url of the Thumbnail
+   */
+  url?: string;
+
+  /**
+   * The web Url of the Thumbnail
+   */
+  thumbnailWebUrl?: string;
+}
+
+/**
+ * Object representing a Search Answer
+ */
+type Answer = {
+  '@odata.type': string;
+  displayName?: string;
+  description?: string;
+  webUrl?: string;
+};
+
+/**
+ * Object representing a search resource supporting thumbnails
+ */
+type ThumbnailResource = {
+  thumbnail: Thumbnail;
+};
+
+type UserResource = {
+  lastModifiedBy?: {
+    user?: {
+      email?: string;
+    };
+  };
+  userPrincipalName?: string;
+};
+
+/**
+ * Object representing a Search Resource
+ */
+type SearchResource = Partial<
+  DriveItem & Site & List & Message & ListItem & Drive & DirectoryObject & Answer & ThumbnailResource & UserResource
+>;
+
+/**
+ * Object representing a full Search Response
+ */
+type SearchResponseCollection = {
+  value?: SearchResponse[];
+};
```

Comment:
Could we do this instead as we have a type to wrap access to the value property in an generic manner?

```suggestion
type SearchResponseCollection = CollectionResponse<SearchResponse>;
```